fdbclient/NativeAPI.actor.cpp

```diff
@@ -2602,19 +2602,19 @@ void Transaction::setupWatches() {
 
 ACTOR static Future<Void> tryCommit( Database cx, Reference<TransactionLogInfo> trLogInfo, CommitTransactionRequest req, Future<Version> readVersion, TransactionInfo info, Version* pCommittedVersion, Transaction* tr, TransactionOptions options) {
 	state TraceInterval interval( "TransactionCommit" );
-	state double startTime;
+	state double startTime = now();
 	if (info.debugID.present())
 		TraceEvent(interval.begin()).detail( "Parent", info.debugID.get() );
 
-	if(CLIENT_BUGGIFY) {
-		throw deterministicRandom()->randomChoice(std::vector<Error>{
-				not_committed(),
-				transaction_too_old(),
-				proxy_memory_limit_exceeded(),
-				commit_unknown_result()});
-	}
-
 	try {
+		if(CLIENT_BUGGIFY) {
+			throw deterministicRandom()->randomChoice(std::vector<Error>{
+					not_committed(),
+					transaction_too_old(),
+					proxy_memory_limit_exceeded(),
+					commit_unknown_result()});
```

Comment:
After looking at this I wondered why database_locked isn't in here, but I guess a user may never intend to lock their database so may not want to test that case.